[Feature] 지난주 지출 리포트 조회 API #48

ockko opened the pull request:
## #️⃣ Issue Number

#48 

## 📝 요약(Summary)

- GET /dashboard/reports/weekly 엔드포인트 생성
- 오늘 날짜를 기준으로 지난주(월요일~일요일)의 시작일과 종료일을 계산하는 로직 구현
- transactions 테이블에서 해당 사용자의 지난주 모든 지출 내역을 조회하는 쿼리 작성
- 조회된 지출 내역을 바탕으로 '요일별 지출액', '일일 평균 지출액', '목표 달성률'을 계산하는 로직 구현
- 계산된 모든 리포트 데이터를 DTO에 담아 반환하도록 API 완성

## 🛠️ PR 유형

어떤 변경 사항이 있나요?

- [x] 새로운 기능 추가
- [ ] 버그 수정
- [ ] CSS 등 사용자 UI 디자인 변경
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [ ] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [x] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## 📸스크린샷 (선택)

## 💬 공유사항 to 리뷰어
`int averageDailySpending = (int) (totalSpending / 7);`
일일 평균 지출액은 내림으로 처리하였습니다.
- 추천 일일 소비 한도 계산 시, 보수적인 기준으로 추천해주기 위해 내림 사용.
- 일일 평균 지출액도 서비스 전체의 일관성을 유지하기 위해 내림으로 처리.
- 사용자가 실제 쓴 돈보다 적게 썼다고 생각할 수 있지만 그 차이는 미비하다고 생각했습니다.
- 34,500.7원 -> 34,500원

## ✅ PR Checklist

PR이 다음 요구 사항을 충족하는지 확인하세요.

- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

ockko committed:
feat: 지난주 지출 리포트 조회 API #48

- GET /dashboard/reports/weekly/latest 엔드포인트 생성
- 오늘 날짜를 기준으로 지난주(월요일~일요일)의 시작일과 종료일을 계산하는 로직 구현
- transactions 테이블에서 해당 사용자의 지난주 모든 지출 내역을 조회하는 쿼리 작성
- 조회된 지출 내역을 바탕으로 '요일별 지출액', '일일 평균 지출액', '목표 달성률'을 계산하는 로직 구현
- 계산된 모든 리포트 데이터를 DTO에 담아 반환하도록 API 완성

kirikaka commented on src/main/java/cheongsan/domain/deposit/controller/ReportController.java:
```diff
@@ -0,0 +1,27 @@
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/cheongsan/dashboard/reports/weekly/latest")
```
추후에 메일 서비스를 진행하려면 RequestMapping을 분리하는것이 좋지 않을까요?
Ex) @RequestMapping("/cheongsan/dashboard")
@GetMapping(/reports/weekly/latest)

kirikaka replied:
확인했습니다.

ockko committed:
refactor: 지난주 지출 리포트 조회 API의 엔드포인트 수정 #48

/cheongsan/dashboard/reports/weekly/latest에서 /cheongsan/dashboard/reports/weekly로 수정
